### ISSUE TITLE
Explain `--filter` option in `ps`

### ISSUE THREAD

**Issue by yariksheptykin** [docker/compose#8643]
The documentation on `--filter` option of `compse ps` is quite sparse. It mentions that the value should be of KEY=VAL form but does not provide any information about what the acceptable keys and values are. One might expect that these key-value pairs can be anything referring to a service spec. Surprizingly both the keys and values are limited to a fixes set of allowed values. Only after looking at the source code and the corresponding acceptance [tests](https://github.com/docker/compose/blob/master/tests/acceptance/cli_test.py#L637-L663) the supported filters are clear.

Ideally help on `ps` command would list out possible key=value pairs. Since the range is currently quite limited it should not be a problem to list all avaliable filters right within help. This in turn can appear in [web documentation](https://docs.docker.com/compose/reference/ps/) and could be consumed by users unable to read compose's sources.

**Describe alternatives you've considered**

Alternatively, command execution itself could list out available filters and options, when invoked with unsupported key=value pairs.
Another option would be to document this option in more detail in the docs. This however risks introducing drift between the documntation and actual iimplemtation if acceptable key=value pairs change.

**Additional context**
I spent quite some time figuring out how to use the filter. So I am creating a PR to help future me get there faster.

**Comment by ndeloof** [comment]
I agree this flag is poorly documented and hard to use for end users as it suppose some knowledge of the available filter attributes. That's why I preferred a more explicit flag in my Compose V2 PR: https://github.com/docker/compose-cli/pull/1823

**Comment by yariksheptykin** [comment]
Thanks for quick feedback, @ndeloof . The approach you take in PR is reasonable as long as the list of filter is so limited. I assume the same can be done to `--source` as well.

So, just to make sure I understand it right, this repo is for `docker-compose` binary (compose v1.x) and the changes I suggest in PR #8644 apply to compose v1.x. Your PR @ndeloof addresses compose command in docker (compose v2.x). Is that right? So, should we update v1.x at all? Or should I rather improve this bit of documentation in v2.x? Or both? I am not familiar with the release roadmap of v2.x to make a reasonable decision. Personally I am using v1.x as long as v2.x remains BETA.

**Comment by ndeloof** [comment]
My PR indeed addressed compose V2, which code has moved into https://github.com/docker/compose/tree/v2

v2.x is Release Candidate, not BETA, and will be "final" in a few.

**Comment by yariksheptykin** [comment]
If I read the [Context of Docker Compose evolution](https://docs.docker.com/compose/cli-command/#context-of-docker-compose-evolution) correctly this issue is still relevant for `docker-compose` (v1) since "docker-compose is still supported and maintained". So, I am leaving this issue and the PR #8644 open as both target v1.

compose v2 does seem to provide much more [documentation](https://github.com/docker/compose/blob/v2/cmd/compose/ps.go#L84-L85) neither for `--filter` nor for `--status`. So, for compose v2 I will create another PR targeting v2 branch of this repo where I could document the options somewhat more.

What confuses me is "compose ps --filter KEY-VALUE Not relevant due to its complicated usage with the service command and also because it is not documented properly in docker-compose." statement from [Flags that will not be implemented](https://docs.docker.com/compose/cli-command-compatibility/#flags-that-will-not-be-implemented)

If the support for `--filter` is going to be dropped anyways, then there is no point in imroving documentation in v2. The bits of information I could pull together are somewhat conflicting. Do you have a suggestion on how we should proceed with this issue regarding compose v2 @ndeloof ?

**Comment by ndeloof** [comment]
On one hand, we want to offer backward compatibility, so I don't think `--filter` will be removed (maybe hidden/deprecated?)
On the other hand, such filter are so hard to use I don't see a major benefit to offer more documentation.
I'd suggest to update `--status` with list of supported status values (which are basically the ones from engine API)

**Comment by yariksheptykin** [comment]
This seems related https://github.com/docker/docker.github.io/issues/9594
